lib/astutils.cpp

```diff
@@ -2407,7 +2407,7 @@ bool isVariableChangedByFunctionCall(const Token *tok, int indirect, const Setti
             continue;
         conclusive = true;
         if (indirect > 0) {
-            if (!arg->isConst() && arg->isPointer())
+            if (arg->isPointer() && !(arg->valueType() && arg->valueType()->pointer == 1 && arg->valueType()->constness == 1))
```

Comment:
This should probably be `arg->valueType()->pointer == indirect && arg->valueType()->constness == indirect`.